woocommerce/woocommerce-ios
# [Coupons in Orders]Validate coupon in orders

Closes: #9999 


## Description

With this PR we enhance the coupons in orders feature by validating the coupon -checking if it exists remotely- before closing the coupon screen and trying to add the coupon to the order. 

## Testing instructions


1. Go to orders
2. Tap on + to create a new order
3. Tap on + Add coupon
4. Enter a coupon code that doesn't exist. It should show a loading indicator and an error.
5. Enter a coupon code that exists. It should add the coupon to the order without any error.
6. Enter a coupon code that exists, but all in uppercase.  It should add the coupon to the order without any error. (For the backend all coupons are lowercase, so we had to transform them locally, as otherwise it returns error)
## Screenshots

### Before
See how the add coupon screen doesn't validate the coupon, and fails with a wrong error description when trying to add it to the order:

https://github.com/woocommerce/woocommerce-ios/assets/1864060/f2d37a09-ea55-4acb-9973-f7735ed4ee12

### After

https://github.com/woocommerce/woocommerce-ios/assets/1864060/e5fa21c7-b368-4969-89ca-05bb22c1b7c6

---
